devtools/cli/src/main/java/io/quarkus/cli/Encryptor.java

```diff
@@ -0,0 +1,60 @@
+package io.quarkus.cli;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.SecureRandom;
+import java.util.Base64;
+import java.util.concurrent.Callable;
+
+import javax.crypto.Cipher;
+import javax.crypto.spec.GCMParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+import io.quarkus.cli.common.OutputOptionMixin;
+import picocli.CommandLine;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+
+@Command(name = "encryptor", header = "Encrypt Secrets using AES/GCM/NoPadding algorithm by default")
+public class Encryptor implements Callable<Integer> {
+    @Option(required = true, names = { "-s", "--secret" }, description = "Secret")
+    String secret;
+
+    @Option(required = true, names = { "-k", "--encryption-key" }, description = "Encryption Key")
+    String encryptionKey;
+
+    @Option(hidden = true, names = { "-a", "--algorithm" }, description = "Algorithm", defaultValue = "AES")
```

Comment:
is there some hints in users config on what algorithm they might be using for this so default value could be figured out from there ?

not a blocker for this pr - just trying to think what could be best in future.

Reply:
Yes, the secret handler expression. For instance, to set a secret value and a way to decrypt it you need something like:

```properties
quarkus.datasource.password=${aes-gcm-nopadding::SECRET}
smallrye.config.secret-handler.aes-gcm-nopadding.encryption-key=key
```

But I'm probably going to change that to be more generic, like:

```properties
quarkus.datasource.password=${dec::SECRET}
smallrye.config.secret-handler.dec.alg=AES
...
```

The only problem is that you can have multiple handlers. We can discover all the available handlers and the user can select which one to use. We can also retrieve the encryption key if available (not likely, because I would expect that the encryption key will not be stored in `application.properties` or any other source available to the CLI, but we can do it as a best effort).